packages/flutter/lib/src/cupertino/segment_control.dart

```diff
@@ -0,0 +1,191 @@
+import 'package:flutter/widgets.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/painting.dart';
+
+class SegmentControlItem {
+  final String title;
+  final Widget content;
+
+  SegmentControlItem(this.title, this.content);
+}
+
+class SegmentControl extends StatefulWidget {
+  final List<SegmentControlItem> tabs;
+  final int activeTabIndex;
+
+  SegmentControl(this.tabs, {this.activeTabIndex = 0})
+      : assert(tabs.length > 1 && tabs.length <= 3), assert(activeTabIndex <= tabs.length - 1);
+
+  @override
+  State createState() {
+    return new _SegmentControlState();
+  }
+}
+
+class _SegmentControlState extends State<SegmentControl> {
+  int _activeTabIndex;
+
+  @override
+  void initState() {
+    super.initState();
+
+    setState(() {
+      _activeTabIndex = widget.activeTabIndex;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    Widget activeTab = widget.tabs[_activeTabIndex].content;
+
+    List<_SegmentControlItem> list = <_SegmentControlItem>[];
+
+    for (int i = 0; i < widget.tabs.length; i++) {
+      SegmentControlItem tap = widget.tabs[i];
+      bool isActive = tap == widget.tabs[_activeTabIndex];
+      _ButtonPlace place = _ButtonPlace.start;
+
+      if (i > 0 && (widget.tabs.length - 1 == i)) {
+        place = _ButtonPlace.end;
+      } else if (i > 0 && (widget.tabs.length - 1 > i)) {
+        place = _ButtonPlace.middle;
+      }
+
+      list.add(new _SegmentControlItem(this, tap, place, isActive));
+    }
+
+    return new Column(
+      mainAxisAlignment: MainAxisAlignment.start,
+      children: <Widget>[
+        new Padding(
+          child: new Row(
+            mainAxisAlignment: MainAxisAlignment.center,
+            children: list,
+          ),
+          padding: new EdgeInsets.all(12.0),
+        ),
+        activeTab
+      ],
+    );
+  }
+
+  void changeTab(String title) {
+    setState(() {
+      for (int i = 0; i < widget.tabs.length; i++) {
+        SegmentControlItem t = widget.tabs[i];
+        if (t.title == title) {
```

Comment:
this is very expensive (O(N*M), where N is the number of widgets and M is the length of the titles), and will fail if any title is duplicated.